sound/soc/sof/loader.c

```diff
@@ -130,6 +130,18 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 }
 EXPORT_SYMBOL(snd_sof_fw_parse_ext_data);
 
+static int ext_man_get_fw_version(struct snd_sof_dev *sdev,
+				  const struct ext_man_elem_header *hdr)
+{
+	const struct ext_man_fw_version *v;
+
+	v = container_of(hdr, struct ext_man_fw_version, hdr);
+
+	memcpy(&sdev->fw_ready, &v->version, sizeof(v->version));
+
+	return snd_sof_ipc_valid(sdev);
```

Comment:
@ktrzcinx Hmm, I know I complained about some obvious comments, but I think here some comments would help. Why the call to snd_sof_ipc_valid()?

Reply:
added